tasks/main.yml

```diff
@@ -38,12 +38,42 @@
         msg: >-
           The mssql_input_sql_file variable is deprecated and will be removed
           in a future version. Edit your playbook to use
-          mssql_post_input_sql_file and mssql_pre_input_sql_file variables
+          mssql_post_input_sql_files and mssql_pre_input_sql_files variables
           instead.
 
     - name: Link the deprecated mssql_input_sql_file fact
       set_fact:
-        mssql_post_input_sql_file: "{{ mssql_input_sql_file }}"
+        mssql_post_input_sql_files: "{{ mssql_input_sql_file }}"
+
+- name: Link the deprecated mssql_pre_input_sql_file fact
+  when: mssql_pre_input_sql_file is defined
+  block:
+    - name: Print that the mssql_pre_input_sql_file variable is deprecated
+      debug:
+        msg: >-
+          The mssql_pre_input_sql_file variable is deprecated and will be
+          removed in a future version. Edit your playbook to use the
+          mssql_pre_input_sql_files (written with files in plural) variable
+          instead.
+
+    - name: Link the deprecated mssql_pre_input_sql_file fact
+      set_fact:
+        mssql_pre_input_sql_files: "{{ mssql_pre_input_sql_file }}"
```

Comment:
If it is possible that the user has specified a list in `mssql_pre_input_sql_file`, then
```yaml
        mssql_pre_input_sql_files: "{{ (mssql_pre_input_sql_file is string) | ternary([mssql_pre_input_sql_file], mssql_pre_input_sql_file) }}"
```
There isn't really a good test for `list` - both `iterable` and `sequence` return `true` for strings - https://jinja.palletsprojects.com/en/2.9.x/templates/#list-of-builtin-tests - so this assumes if it is not a string, it is a list

Reply:
Hi Rich, please take a look at the last commit `Allow *_input_sql_files vars to take lists and strings`, do you think if making var to take both lists and strings makes sense?

Reply:
> Hi Rich, please take a look at the last commit `Allow *_input_sql_files vars to take lists and strings`, do you think if making var to take both lists and strings makes sense?

Looks good